openedx/core/djangoapps/credit/email_utils.py

```diff
@@ -180,3 +206,60 @@ def _email_url_parser(url_name, extra_param=None):
     dashboard_url_path = reverse(url_name) + extra_param if extra_param else reverse(url_name)
     dashboard_link_parts = ("https", site_name, dashboard_url_path, '', '', '')
     return urlparse.urlunparse(dashboard_link_parts)
+
+
+def get_credit_providers_by_course(user, course_key):
+    """Get the course information from ecommerce and parse the data to get providers.
+
+    Arguments:
+        user (User): The user object.
+        course_key (CourseKey): The identifier for the course.
+
+    Returns:
+        list, containing course providers.
+    """
+    credit_config = CreditConfig.current()
+
+    # Bypass caching for staff users, who may be adding providers and want
+    # to see them immediately.
+
+    providers_data = None
+    cache_key = None
+    if credit_config.enabled:
+        cache_key = '{key_prefix}.{course_key}'.format(
+            key_prefix=credit_config.CACHE_KEY, course_key=unicode(course_key)
+        )
+        providers_data = cache.get(cache_key)
+
+    if providers_data is not None:
+        return providers_data
+
+    providers_data = []
+    try:
+        response = ecommerce_api_client(user).courses(unicode(course_key) + '/?include_products=1').get()
+    except Exception:  # pylint: disable=broad-except
+        log.exception("Failed to receive data from the ecommerce course API for Course ID '%s'.", unicode(course_key))
+        return
+
+    if not response:
+        log.debug("No Course information found from ecommerce API for Course ID '%s'.", unicode(course_key))
+        return
+
+    for product in response.get('products'):
+        providers_data.extend(
+            [attr.get('value') for attr in product.get('attribute_values') if attr.get('name') == 'credit_provider']
+        )
+
+    # Hit the CreditProvider method to get the all active providers with complete information.
+    providers_information = CreditProvider.get_credit_providers()
+
+    providers_display_names = []
+    for provider_id in providers_data:
+        providers_display_names.extend(
+            [provider['display_name'] for provider in providers_information if provider['id'] == provider_id]
+        )
+
+    if cache_key:
```

Comment:
Use `is_cache_enabled`. The presence of the cache key is not sufficient when deciding whether to use the cache.